src/Servers/IIS/IntegrationTesting.IIS/src/IISDeployer.cs

```diff
@@ -473,6 +490,26 @@ private static void RetryServerManagerAction(Action<ServerManager> action)
             delay *= 1.5;
         }
 
+        // Try to upload the applicationHost config on helix to help debug
+        UploadConfigFiles("serverManagerRetryFailed");
+
         throw new AggregateException($"Operation did not succeed after {retryCount} retries", exceptions.ToArray());
     }
+
+    // This will put the files in your user temp folder when running locally
+    private static void UploadFileOnHelix(string filePath, string uploadFileName)
+    {
+        var HELIX_WORKITEM_UPLOAD_ROOT = Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT");
```

Comment:
Please don't use an all-caps variable name e.g.
```suggestion
        var uploadRoot = Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT");
```